Makefile

```diff
@@ -315,9 +315,40 @@ ifeq ('$(browser)','chrome-mv3')
   BUILD_TARGETS += $(BUILD_DIR)/data/bundled/smarter-encryption-rules.json
 endif
 
+## Generate Tracker Blocking declarativeNetRequest rulesets for MV3 builds.
+MV3_BLOCKLIST_TARGETS = $(INTERMEDIATES_DIR)/current-mv3-tds.json $(INTERMEDIATES_DIR)/fallback-mv3-tds.json
+MV3_CURRENT_RULESET_TARGETS = $(BUILD_DIR)/data/bundled/current-tds-rules.json $(BUILD_DIR)/data/bundled/current-ctl-allowing-rules.json
+MV3_FALLBACK_RULESET_TARGETS = $(BUILD_DIR)/data/bundled/fallback-tds-rules.json $(BUILD_DIR)/data/bundled/fallback-ctl-allowing-rules.json
+
+.SECONDARY:
+$(MV3_BLOCKLIST_TARGETS) &: shared/data/etags.json
+	node scripts/fetchMV3Blocklists.mjs
+	touch $(MV3_BLOCKLIST_TARGETS)
+
+$(BUILD_DIR)/data/bundled/current-mv3-tds.json: $(INTERMEDIATES_DIR)/current-mv3-tds.json
+	cp $< $@
+
+$(BUILD_DIR)/data/bundled/fallback-mv3-tds.json: $(INTERMEDIATES_DIR)/fallback-mv3-tds.json
+	cp $< $@
+
+$(MV3_CURRENT_RULESET_TARGETS) &: $(INTERMEDIATES_DIR)/current-mv3-tds.json $(INTERMEDIATES_DIR)/surrogates.json
```

Comment:
I'm not sure what the `&` does here, but it spits out a load of warnings for me when I run `make` (on a mac):
```
Makefile:335: warning: overriding commands for target `&'
Makefile:325: warning: ignoring old commands for target `&'
Makefile:338: warning: overriding commands for target `&'
Makefile:335: warning: ignoring old commands for target `&'
```

Reply:
The `&:` syntax is for a [rule with grouped target](https://www.gnu.org/software/make/manual/html_node/Multiple-Targets.html#Rules-with-Grouped-Targets). It's used for when one command is used to generate multiple target files at the same time, when used Make knows to only run the command once to generate all the files.

I will try to debug what's going wrong. What version of Make are you running?

Reply:
```bash
$ make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
```

Reply:
Seems that support for the `&:` syntax was added with 4.3, but macOS is stuck with an outdated version of Make due to licensing 🤦. I have avoided that by using [the empty target trick](https://www.gnu.org/software/make/manual/html_node/Empty-Targets.html). Is that working better for you now?